paddle/fluid/operators/fc_op.cc

```diff
@@ -142,13 +142,12 @@ class FCOpMaker : public framework::OpProtoAndCheckerMaker {
     AddAttr<bool>(framework::kAllKernelsMustComputeRuntimeShape,
                   "Skip calling InferShape() function in the runtime.")
         .SetDefault(true);
+    AddAttr<std::string>(
+        "mkldnn_data_type",
+        "(string, default \"float32\"). Data type of mkldnn kernel")
+        .SetDefault("float32")
+        .InEnum({"float32", "int8", "bfloat16"});
```

Comment:
I just wonder does it have to be string?
enum class MKLDNN_DATATYPE{ FLOAT32, INT8, BFLOAT16};

Reply:
I think using enum is hard because of Python-C++ compatibility of attribute type. This is why most operators use strings with `InEnum()` check.